include/server/api/base_parameters_grammar.hpp

```diff
@@ -88,21 +88,32 @@ struct BaseParametersGrammar : boost::spirit::qi::grammar<Iterator, Signature>
             }
             else
             {
-                base_parameters.hints.emplace_back(boost::none);
+                base_parameters.hints.emplace_back(std::nullopt);
             }
         };
 
         const auto add_bearing =
             [](engine::api::BaseParameters &base_parameters,
                boost::optional<boost::fusion::vector2<short, short>> bearing_range)
         {
-            boost::optional<engine::Bearing> bearing;
+            std::optional<engine::Bearing> bearing;
             if (bearing_range)
             {
                 bearing = engine::Bearing{boost::fusion::at_c<0>(*bearing_range),
                                           boost::fusion::at_c<1>(*bearing_range)};
             }
-            base_parameters.bearings.push_back(std::move(bearing));
+            base_parameters.bearings.push_back(bearing);
+        };
+
+        const auto add_approach = [](engine::api::BaseParameters &base_parameters,
+                                     boost::optional<osrm::engine::Approach> approach) {
```

Comment:
These 2 lambdas in this grammar is single place where we have boost::optional left - just because Boost Spirit doesn't work well with std::optional, so we use boost::optional here and immediately convert it to std::optional